src/managed/Microsoft.Extensions.DependencyModel/DependencyContextWriter.Netstandard2_0.cs

```diff
@@ -0,0 +1,442 @@
+                jsonWriter.WriteString(DependencyContextStrings.LocalePropertyName, resourceAssembly.Locale, escape: false);
+                jsonWriter.WriteEndObject();
+            }
+            jsonWriter.WriteEndObject();
+        }
+
+        private void WriteTargetLibrary(string key, Library library, ref Utf8JsonWriter jsonWriter)
+        {
+            if (library is RuntimeLibrary runtimeLibrary)
+            {
+                jsonWriter.WriteStartObject(key);
+
+                AddDependencies(runtimeLibrary.Dependencies, ref jsonWriter);
+
+                // Add runtime-agnostic assets
+                AddAssets(DependencyContextStrings.RuntimeAssembliesKey, runtimeLibrary.RuntimeAssemblyGroups.GetDefaultGroup(), ref jsonWriter);
+                AddAssets(DependencyContextStrings.NativeLibrariesKey, runtimeLibrary.NativeLibraryGroups.GetDefaultGroup(), ref jsonWriter);
+                AddResourceAssemblies(runtimeLibrary.ResourceAssemblies, ref jsonWriter);
+
+                jsonWriter.WriteEndObject();
+            }
+            else if (library is CompilationLibrary compilationLibrary)
+            {
+                jsonWriter.WriteStartObject(key);
+                AddDependencies(compilationLibrary.Dependencies, ref jsonWriter);
+                AddCompilationAssemblies(compilationLibrary.Assemblies, ref jsonWriter);
+                jsonWriter.WriteEndObject();
+            }
+            else
+            {
+                throw new NotSupportedException();
+            }
+        }
+
+        private void WritePortableTargetLibrary(string key, RuntimeLibrary runtimeLibrary, CompilationLibrary compilationLibrary, ref Utf8JsonWriter jsonWriter)
+        {
+            jsonWriter.WriteStartObject(key);
+
+            var dependencies = new HashSet<Dependency>();
+            if (runtimeLibrary != null)
+            {
+                dependencies.UnionWith(runtimeLibrary.Dependencies);
+            }
+
+            if (compilationLibrary != null)
+            {
+                dependencies.UnionWith(compilationLibrary.Dependencies);
+            }
+            AddDependencies(dependencies, ref jsonWriter);
+
+
+            if (runtimeLibrary != null)
+            {
+                // Add runtime-agnostic assets
+                AddAssets(DependencyContextStrings.RuntimeAssembliesKey, runtimeLibrary.RuntimeAssemblyGroups.GetDefaultGroup(), ref jsonWriter);
+                AddAssets(DependencyContextStrings.NativeLibrariesKey, runtimeLibrary.NativeLibraryGroups.GetDefaultGroup(), ref jsonWriter);
+                AddResourceAssemblies(runtimeLibrary.ResourceAssemblies, ref jsonWriter);
+
+                // Add runtime-specific assets
+                bool wroteObjectStart = false;
+                wroteObjectStart = AddRuntimeSpecificAssetGroups(DependencyContextStrings.RuntimeAssetType, runtimeLibrary.RuntimeAssemblyGroups, wroteObjectStart, ref jsonWriter);
```

Comment:
This was necessary since the writer is forward only (while building a JObject and writing at the end did not care about the order).

Reply:
probably worth a comment in the code then until this moves to the JsonDocument (writer)

Reply:
I don't think we would move this to JsonDocument (as it is written today) especially since this is writing (not reading). I was only suggesting that we needed this `bool wroteObjectStart` due to the ordering of the writes being important (while the ordering of adding to `JObject` is not really important).